internal/k8s/reconciler/gateway.go

```diff
@@ -74,27 +39,6 @@ func (g *K8sGateway) ID() core.GatewayID {
 	}
 }
 
-// Bind returns the name of the listeners to which a route bound
-// TODO Remove
-func (g *K8sGateway) Bind(ctx context.Context, route store.Route) []string {
```

Comment:
This logic and the related logic below is now handled elsewhere and no longer needed since we're removing the `StatusTrackingGateway` interface